### ISSUE TITLE
Implement hextra/hero-section, defining a h2 heading on front-page

### ISSUE THREAD

**Issue by fmunteanu** [imfing/hextra#388]
**Feature Description**

Hi @imfing,

On Hextra front-page, I need the ability to insert a `h2` heading, defining additional section titles. For example, adding a subtitle above `hextra/feature-grid`.

**Proposed Solution**

I created the following `layouts/shortcodes/hextra/hero-section.html` shortcode, implementing a section:

```
{{- $style := .Get "style" -}}

<h2
  class="not-prose hx-text-4xl hx-font-bold hx-leading-none hx-tracking-tighter md:hx-text-3xl hx-py-2 hx-bg-clip-text hx-text-transparent hx-bg-gradient-to-r hx-from-gray-900 hx-to-gray-600 dark:hx-from-gray-100 dark:hx-to-gray-400"
  {{ with $style }}style="{{ . | safeCSS }}"{{ end }}
>
  {{ .Inner | markdownify }}
</h2>
```

**Code Usage**

Example of `hextra/hero-section` shortcode usage:

```
<div class="hx-mt-6 hx-mb-6">
{{< hextra/hero-section >}}
  Used Technologies
{{< /hextra/hero-section >}}
</div>
```

**Implementation Result**

Note the `Used Technologies` section title, above the `hextra/feature-grid`:

![image](https://github.com/imfing/hextra/assets/19806136/f01b86d5-d29f-4ce6-8f88-e94d42678c98)

**Comment by mafendi** [comment]
Hi! 

Could you please provide a code that you use for a big picture in a hero section?

Thanks.

**Comment by fmunteanu** [comment]
@mafendi see #387.

**Comment by mafendi** [comment]
> @mafendi see #387.

Thanks! 😊

**Comment by fmunteanu** [comment]
I think these two hero features are very useful for the front-page and should be implemented.

**Comment by imfing** [comment]
@fmunteanu Can you send PRs for these? thank you

**Comment by fmunteanu** [comment]
@imfing will do, thank you for considering it!

**Comment by fmunteanu** [comment]
@imfing I implemented header size in #390. Is a nice addition to original feature.

**Comment by fmunteanu** [comment]
@imfing can we have this PR merged into next release?
